testsuite/libra-fuzzer/src/fuzz_targets/storage_save_blocks.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::FuzzTargetImpl;
+use libra_proptest_helpers::ValueGenerator;
+use libradb::{test_helper::arb_blocks_to_commit, test_save_blocks_impl};
+use proptest::{
+    strategy::{Strategy, ValueTree},
+    test_runner::{self, TestRunner},
+};
+use rand::RngCore;
+
+#[derive(Clone, Debug, Default)]
+pub struct StorageSaveBlocks;
+
+impl FuzzTargetImpl for StorageSaveBlocks {
+    fn name(&self) -> &'static str {
+        module_name!()
+    }
+
+    fn description(&self) -> &'static str {
+        "Storage save blocks"
+    }
+
+    fn generate(&self, _idx: usize, _gen: &mut ValueGenerator) -> Option<Vec<u8>> {
+        let mut output = vec![0u8; 4096];
+        let mut rng = rand::thread_rng();
+        rng.fill_bytes(&mut output);
+        Some(output)
```

Comment:
Hmm, so does this generate any arbitrary sequence of bytes? Are all such sequences valid?

Reply:
yes (uses thread_rng) and yes (any random output is valid for a strategy), but a better question is: is this enough or too much output for the strategy?

For now I have no idea. 

A better way would be to ask proptest to generate a bunch of values, and then dump the  random bytes that it used to generate them, unfortunately there doesn't seem to be any API for that.

Some ideas:

* push a PR upstream to create this API
* generate a value using this `output` with proptest
  - if it couldn't create a value, try again with a random output twice the size (and again and again)
  - if it worked, see how much bytes are left in the `output` and discard them. 

what do you think?

Reply:
Would the pass through API work for most of this?

Reply:
I don't think so, since we can't know how much of the seed was used. If the passthrough crashed (the behavior before my patch) we could do something sketchy where we generate a seed of length X, catch a panic, start again with length X+1, and so on.